contk/dataloader/dataloader.py

```diff
@@ -26,3 +30,255 @@ def load_class(cls, class_name):
 			if subclass.__name__ == class_name:
 				return subclass
 		return None
+
+
+class BasicLanguageGeneration(Dataloader):
+	'''Base class'''
+	def __init__(self, \
+				 index_key, \
+				 ext_vocab=None, \
+				 key_name=None):
+		super().__init__()
+
+		# initialize by default value. (can be overwritten by subclass)
+		self.ext_vocab = ext_vocab or ["<pad>", "<unk>", "<go>", "<eos>"]
+		self.pad_id = self.ext_vocab.index("<pad>")
+		self.unk_id = self.ext_vocab.index("<unk>")
+		self.go_id = self.ext_vocab.index("<go>")
+		self.eos_id = self.ext_vocab.index("<eos>")
+		self.key_name = key_name or ["train", "dev", "test"]
+
+		# initialize by subclass
+		self.vocab_list, self.data = self._load_data()
+		self.word2id = {w: i for i, w in enumerate(self.vocab_list)}
+
+		# postprocess initialization
+		self.index = {}
+		self.batch_id = {}
+		self.batch_size = {}
+		for key in self.key_name:
+			self.batch_id[key] = 0
+			self.batch_size[key] = None
+			self.index[key] = list(range(len(self.data[key][index_key])))
```

Comment:
`index_key` is not easy to understand. Make a change to `_load_data`?

Reply:
I think is not convenient to change `_load_data`, unless we change the three different keys (`'sen','session','post'`) in `language_generation`, `multi_turn_dialog`, `single_turn_dialog`

Reply:
Let `_load_data` to return a dict containing the size of index